src/composed/signed_key/shared.rs

```diff
@@ -246,6 +247,41 @@ impl PublicOrSecret {
     }
 }
 
+/// Error returned when trying to convert `PublicOrSecret` key
+/// into the wrong type.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct TryFromPublicOrSecretError;
+
+impl fmt::Display for TryFromPublicOrSecretError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Attempt to convert PublicOrSecret key to the wrong type")
+    }
+}
+
+impl std::error::Error for TryFromPublicOrSecretError {}
+
+impl TryFrom<PublicOrSecret> for SignedPublicKey {
+    type Error = TryFromPublicOrSecretError;
```

Comment:
Wouldn't it be easier and more ergonomic to use `pgp::errors::Error` here?

Reply:
Hmm. I never thought about how big `pgp::errors::Error` is, in terms of memory. I would have hoped it's minor.

Is there reason to consider this operation performance critical? The parsing that precedes this call should be more expensive by orders of magnitude.

I'd worry that a separate error type makes error handling more confusing for users of rPGP.

So my preference would be to reuse the regular rpgp error type, but this preference is not very strongly held.

Reply:
we probably need to split that error up into multiple suberrors anyway at some point, so this fine